### DISCUSSION TITLE
why does useDisclosure not throw "Invalid hook call" error?

### DISCUSSION THREAD

**Discussion by NoahFletcherUWM** [chakra-ui/chakra-ui#6777] [Q&A]
I was wondering how [this](https://github.com/chakra-ui/chakra-ui/blob/main/packages/hooks/use-disclosure/src/index.ts#:~:text=%7D-,handleClose%3F.(),-%7D%2C%20%5B) and [this](https://github.com/chakra-ui/chakra-ui/blob/main/packages/hooks/use-disclosure/src/index.ts#:~:text=%7D-,handleOpen%3F.(),-%7D%2C%20%5B) are not throwing that one error that goes like... `Uncaught Error: Invalid hook call. Hooks can only be called inside of the body of a function component.` etc... 
Why is it allowing another hook to be used inside the parent `useCallback` hook?

If somebody could explain what I'm missing id appreciate it.

Edit: `this` and `this` are line 44 and 51*

[useCallbackRef](https://github.com/chakra-ui/chakra-ui/blob/3a87f68474221ed402fe5d169f7067763c8436d5/packages/legacy/hooks/src/use-callback-ref.ts) and [useSafeLayoutEffect](https://github.com/chakra-ui/chakra-ui/blob/3a87f68474221ed402fe5d169f7067763c8436d5/packages/legacy/hooks/src/use-safe-layout-effect.ts) incase you need to look at them.

**Comment by clement-lta** [comment]
@NoahFletcherUWM Totally I still don't get your question. but this can help you on a part
Because useCallbackRef is return a function that is memorized by a useCallback. So why do you expect an error?
